File: Current/Product/Production/Common/Common.Tasks/TextElement.vb
Option Explicit On 
Option Strict On

Imports NAnt.Core
Imports NAnt.Core.Attributes

<ElementName("text")> _
Public Class TextElement
    Inherits Element

    Private _Value As String
    Private _Xml As Boolean = False
    Private _Expand As Boolean = False

    <TaskAttribute("xml", Required:=False), BooleanValidator()> _
    Public Property Xml() As Boolean
        Get
            Return _Xml
        End Get
        Set(ByVal value As Boolean)
            _Xml = Value
        End Set
    End Property

    <TaskAttribute("expand", Required:=False), BooleanValidator()> _
    Public Property Expand() As Boolean
        Get
            Return _Expand
        End Get
        Set(ByVal value As Boolean)
            _Expand = value
        End Set
    End Property

    Public ReadOnly Property [Value]() As String
        Get
            If Me._Value Is Nothing Then
                If Me.Xml Then
                    Me._Value = Me.XmlNode.InnerXml
                Else
                    Me._Value = Me.XmlNode.InnerText
                End If
                If Me.Expand Then
                    Me._Value = Me.Project.ExpandProperties(Me._Value, Me.Location)
                End If
            End If
            Return Me._Value
        End Get
    End Property

    Protected Overrides ReadOnly Property CustomXmlProcessing() As Boolean
        Get
            Return True
        End Get
    End Property

    Protected Overrides Sub InitializeXml(ByVal elementNode As System.Xml.XmlNode, ByVal properties As NAnt.Core.PropertyDictionary, ByVal framework As NAnt.Core.FrameworkInfo)
        MyBase.InitializeXml(elementNode, properties, framework)
    End Sub

End Class

File: Product/Production/Nant/CIFactory.NAnt.Tasks/Types/TextElement.cs
using System;
using System.IO;
using System.Text;
using System.Xml;
using System.Collections;
using System.Collections.Generic;
using NAnt.Core;
using NAnt.Core.Attributes;
using System.Text.RegularExpressions;

namespace CIFactory.NAnt.Types
{

    [ElementName("text")]
    public class TextElement : Element
    {
        #region Fields

        private bool _Expand = false;

        private string _Value;

        private bool _Xml = false;

        #endregion

        #region Properties

        protected override bool CustomXmlProcessing
        {
            get { return true; }
        }

        [TaskAttribute("expand", Required = false), BooleanValidator()]
        public bool Expand
        {
            get { return _Expand; }
            set { _Expand = value; }
        }

		public string Value
		{
			get
			{
				if (this._Value == null)
				{
					if (this.Xml)
					{
						string CleanValue = Regex.Replace(this.XmlNode.InnerXml, @"xmlns=""http://nant\.sf\.net/schemas/nant\.xsd""", "");
						this._Value = CleanValue;
					}
					else
					{
						this._Value = this.XmlNode.InnerText;
					}
					if (this.Expand)
					{
						this._Value = this.Project.ExpandProperties(this._Value, this.Location);
					}
				}
				return this._Value;
			}
		}

        [TaskAttribute("xml", Required = false), BooleanValidator()]
        public bool Xml
        {
            get { return _Xml; }
            set { _Xml = value; }
        }

        #endregion

        #region Protected Methods

        protected override void InitializeXml(System.Xml.XmlNode elementNode, PropertyDictionary properties, FrameworkInfo framework)
        {
            base.InitializeXml(elementNode, properties, framework);
        }

        #endregion
    }

}